File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.cpp
class Solution {
public:
    vector<int> findDisappearedNumbers(vector<int>& nums) {
        
        //multiply all indexes of nums with -1
        for(int n: nums){
            int i = abs(n) - 1;
            nums[i] = -1*abs(nums[i]);
        }
        
        vector<int> result;
        for(int i=0;i<nums.size();i++){
            if(nums[i] > 0){
                result.push_back(i+1);
            }
        }
        return result;
        
        
    }
};

/*
int main() {
    vector<int> nums = {4, 3, 2, 7, 8, 2, 3, 1};
    Solution solution;
    vector<int> result = solution.findDisappearedNumbers(nums);
    
    for (int num : result) {
        cout << num << " ";
    }
    cout << endl;

    return 0;
}
*/

File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.py
#time: o(n)
#space : o(n)
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        
        # multiply all values of nums with -1
        for n in nums:
            
            i = abs(n) -1
            nums[i] = -1*abs(nums[i])
        
        result = []
        for i , n  in enumerate(nums):
            if(n > 0):
                result.append(i+1)
                
        return result
        
        
#ums = [4, 3, 2, 7, 8, 2, 3, 1]
#olution = Solution()
#esult = solution.findDisappearedNumbers(nums)
#rint(result)
#       